### PR TITLE
Refactored Gloo and MPI components into separate subdirectories, reduced INFO logs to DEBUG, and fixed op order for Hierarchical Allreduce

### PR DESCRIPTION
**Pull request by tgaddair**

### CODE REVIEW

**Comment by tgaddair** [comment]
@romerojosh this change addresses the Hierarchical Allreduce issue.

File: horovod/common/controller.h [hunk 1]
```diff
@@ -26,15 +26,11 @@
...
-#if HAVE_GLOO
-#include "gloo_context.h"
-#endif
-
 namespace horovod {
 namespace common {
 
 // Forward declaration
-class HorovodGlobalState;
+struct HorovodGlobalState;
```
Comment (alsrgv) [inline horovod/common/controller.h]: Do we still need it here?

File: horovod/common/gloo/gloo_controller.cc [hunk 2]
```diff
@@ -13,20 +13,20 @@
 // limitations under the License.
 // =============================================================================
 
-#include <cstring>
-
 #include "gloo_controller.h"
```
Comment (alsrgv) [inline horovod/common/gloo/gloo_controller.cc]: Move gloo_controller.h after system includes?

File: horovod/common/gloo/http_store.h [hunk 3]
```diff
@@ -13,8 +13,8 @@
 // limitations under the License.
 // ============================================================================
 
-#ifndef HOROVOD_RENDEZVOUS_HTTP_RENDEZVOUS_H_
-#define HOROVOD_RENDEZVOUS_HTTP_RENDEZVOUS_H_
+#ifndef HOROVOD_RENDEZVOUS_HTTP_STORE_H_
```
Comment (alsrgv) [inline horovod/common/gloo/http_store.h]: HOROVOD_GLOO_HTTP_STORE_H

File: horovod/common/controller.h (hunk 1 above)
Comment (tgaddair) [reply horovod/common/controller.h]: Apparently not.  Removed!

File: horovod/common/gloo/gloo_controller.cc (hunk 2 above)
Comment (tgaddair) [reply horovod/common/gloo/gloo_controller.cc]: As discussed, will standardize header include order in a followup PR.

**Pull request merged**